### PR TITLE
Refactored copy methods

### PR DESCRIPTION
**Pull request by ZNick1982**
Refactored copy methods:
- Added File::createLocalCopy and File::createRemoteCopy
- Reafactored Api::copyFile
- Refactored File::copy and File::copyTo methods
- Added unit tests for File::createLocalCopy

### CODE REVIEW

File: src/Uploadcare/File.php [hunk 1]
```diff
@@ -164,6 +164,38 @@ public function copyTo($target)
...
+   * @return File|string
+   */
+  public function createLocalCopy($store = true) {
+    return $this->api->createLocalCopy($this->getUrl(), $store);
+  }
+
+  /**
+   * Copy file to the external storage
+   *
+   * @param string $target Name of custom storage connected to your project. Uploadcare storage is used if target is absent.
```
Comment (dmitry-mukhin) [inline src/Uploadcare/File.php]: second sentence is false

File: src/Uploadcare/File.php [hunk 2]
```diff
@@ -164,6 +164,38 @@ public function copyTo($target)
...
+   * @param boolean $make_public (Optional) MUST be either true or false. true to make copied files available via public links. false to reverse the behavior.
+   * @param string $pattern (Optional) Applies to custom storage usage scenario only. The parameter is used to specify file names Uploadcare passes to custom storages. In case parameter is omitted, custom storage pattern is used.
```
Comment (dmitry-mukhin) [inline src/Uploadcare/File.php]: 1st sentence is redundant, this is only about custom storages

File: src/Uploadcare/File.php (hunk 1 above)
Comment (ZNick1982) [inline src/Uploadcare/File.php]: fixed

File: src/Uploadcare/File.php (hunk 2 above)
Comment (ZNick1982) [inline src/Uploadcare/File.php]: Incorrect sentence removed

**Pull request merged**